app/migrations/20.php

```diff
@@ -0,0 +1,40 @@
+<?php
+  function countUnread($id) {
+    global $MMessage;
+    $messages = array_reverse(iterator_to_array(
+        $MMessage->findByParticipant($id->{'$id'})));
```

Comment:
For all our new code, we should try to use the class :: syntax as much as possible (eg. here would be MessageModel::findByParticipant)